File: src/main/java/org/pyjjs/scheduler/core/api/impl/actors/resource/messages/OfferMessage.java
package org.pyjjs.scheduler.core.api.impl.actors.resource.messages;

import akka.actor.ActorRef;
import org.pyjjs.scheduler.core.api.impl.actors.common.messages.Message;
import org.pyjjs.scheduler.core.model.Resource;

public class OfferMessage extends Message {

    private Double placingPrice = 0d;
    private Resource resource;

    public OfferMessage(ActorRef sender) {
        super(sender);
    }

    public Double getPlacingPrice() {
        return placingPrice;
    }

    public void setPlacingPrice(Double placingPrice) {
        this.placingPrice = placingPrice;
    }

    public Resource getResource() {
        return resource;
    }

    public void setResource(Resource resource) {
        this.resource = resource;
    }
}

File: src/main/kotlin/org/pyjjs/scheduler/core/api/impl/actors/resource/messages/OfferMessage.kt
package org.pyjjs.scheduler.core.api.impl.actors.resource.messages

import akka.actor.ActorRef
import org.pyjjs.scheduler.core.api.impl.actors.common.messages.Message
import org.pyjjs.scheduler.core.model.Resource

class OfferMessage(sender: ActorRef) : Message(sender) {

    var placingPrice: Double = 0.0
    var resource: Resource? = null
}
